vernac/comInductive.ml

```diff
@@ -379,6 +379,14 @@ let interp_mutual_inductive (paramsl,indl) notations cum poly prv finite =
   let arities = List.map nf arities in
   let constructors = List.map (fun (idl,cl,impsl) -> (idl,List.map nf cl,impsl)) constructors in
   let ctx_params = List.map Termops.(map_rel_decl (EConstr.to_constr sigma)) ctx_params in
+  (* Only keep useful universes *)
+  let merge_universes_of_constr c =
+    Univ.LSet.union (EConstr.universes_of_constr sigma (EConstr.of_constr c)) in
+  let uvars = Univ.LSet.empty in
+  let uvars = Context.Rel.(fold_outside (Declaration.fold_constr merge_universes_of_constr) ctx_params ~init:uvars) in
+  let uvars = List.fold_right merge_universes_of_constr arities uvars in
+  let uvars = List.fold_right (fun (_,ctypes,_) -> List.fold_right merge_universes_of_constr ctypes) constructors uvars in
+  let sigma = Evd.restrict_universe_context sigma uvars in
```

Comment:
wow that looks like a very large amount of code to very inlined here, shouldn't it factored out elsewhere?